### PR TITLE
[Pygen]: Add README.md for PyFlow

### PR DESCRIPTION
**Pull request by aneels3**
Signed-off-by: aneels3 <b150023ec@nitsikkim.ac.in>

### CODE REVIEW

File: pygen/pygen_src/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+## Overview
+
+RISCV-DV-PyFlow is a python based open-source instruction generator for RISC-V
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:3]: a purely python based

File: pygen/pygen_src/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+processor verification. It currently supports the following features:
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:4]: We should also talk about / refer to pyvsc as the main tool/library that we're using.

File: pygen/pygen_src/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+- Supported instruction set: RV32IMAFDC
+- Supported privileged mode: machine mode
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:7]: Supported privileged modes: for now only machine mode is supported.

File: pygen/pygen_src/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+- Illegal instruction and HINT instruction generation
+- Random forward/backward branch instructions
+- Supports mixing directed instructions with random instruction stream
+- Support for direct & vectored interrupt table.
+- Multi-hart support
+- Instruction generation coverage model(Currently supports RV32I and RV32M only)
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:13]: Functional coverage framework (reports GUI as well as text, currently supports RV32I and RV32M only)

File: pygen/pygen_src/README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+- Co-simulation with multiple ISS : spike, riscv-ovpsim
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:14]: Supported ISS: Spike, Ovpsim

File: pygen/pygen_src/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+## Overview
+
+RISCV-DV-PyFlow is a python based open-source instruction generator for RISC-V
+processor verification. It currently supports the following features:
+
+- Supported instruction set: RV32IMAFDC
+- Supported privileged mode: machine mode
+- Illegal instruction and HINT instruction generation
+- Random forward/backward branch instructions
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:9]: loops are also supported, right? Mind adding it here?

File: pygen/pygen_src/README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+- riscv_single_hart_test
+- riscv_non_compressed_instr_test
+
+
+## Getting Started
+
+### Prerequisites
+
+To be able to run the generator, you need to have RISCV-GCC compiler toolchain and ISS
+(Instruction Set Simulator) installed.
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:39]: (Instruction Set Simulator) installed (Spike is preferred)

File: pygen/pygen_src/README.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+```bash
+python3 run.py --test=riscv_arithmetic_basic_test --iterations=10 --simulator=pyflow
+```
+Run the generator only, do not compile and simluation with ISS
+```bash
+python3 run.py --test=riscv_arithmetic_basic_test --simulator=pyflow --steps gen
+```
+
+## Note
+Time to generate more than 10k instructions for single iteration is around 10-12 minutes.
```
Comment (hodjat91) [inline pygen/pygen_src/README.md:72]: Currently, time to generate a single program with larger than 10k instructions is around x minutes (let's be more precise, maybe 10-11?). We are working on improving the overall performance.

**Review by hodjat91 — COMMENTED** [review]
Hi @aneels3 

Thanks for the CL. PTAL at the comments and let me know if there is any questions.

File: pygen/pygen_src/README.md (hunk 2 above)
Comment (aneels3) [reply pygen/pygen_src/README.md]: Good point.
I have added it with the pyvsc repo link.

**Comment by aneels3** [comment]
> Hi @aneels3
> 
> Thanks for the CL. PTAL at the comments and let me know if there is any questions.

Hi @hodjat91 
I have addressed all the comments.
Please check it out.
Let me know if it needs more modification.

File: pygen/pygen_src/README.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
+## Overview
+
+RISCV-DV-PyFlow is a purely python based open-source instruction generator for RISC-V processor
```
Comment (udinator) [inline pygen/pygen_src/README.md:3]: nit: capitalize `python` here and throughout

**Review by hodjat91 — APPROVED** [review]
LGTM, please address Udi's comment.

File: pygen/pygen_src/README.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
+## Overview
+
+RISCV-DV-PyFlow is a purely PYTHON based open-source instruction generator for RISC-V processor
```
Comment (udinator) [inline pygen/pygen_src/README.md:3]: nit: sorry for the confusion, I meant `python` -> `Python`

Comment (aneels3) [reply pygen/pygen_src/README.md]: oops!
Fixed now

**Review by udinator — APPROVED** [review]
LGTM, thanks @aneels3 !

**Pull request merged**